From ee3033d8090b6b065c76e8d93c966a31478c5eae Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:58:53 -0400
Subject: [fs] btrfs: fix max dir item size calculation

Message-id: <1407179109-13714-47-git-send-email-sandeen@redhat.com>
Patchwork-id: 86990
O-Subject: [RHEL7.1 PATCH 046/422] Btrfs: fix max dir item size calculation
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Filipe David Borba Manana <fdmanana@gmail.com>

commit 878f2d2cb355da2dabbffb2ae51b7541a91ce4e3
Author: Filipe David Borba Manana <fdmanana@gmail.com>
Date:   Wed Nov 27 16:06:16 2013 +0000

    Btrfs: fix max dir item size calculation

    We were accounting for sizeof(struct btrfs_item) twice, once
    in the data_size variable and another time in the if statement
    below.

    Signed-off-by: Filipe David Borba Manana <fdmanana@gmail.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/dir-item.c b/fs/btrfs/dir-item.c
index c031ea3..9a89ceb 100644
--- a/fs/btrfs/dir-item.c
+++ b/fs/btrfs/dir-item.c
@@ -261,7 +261,7 @@ int btrfs_check_dir_item_collision(struct btrfs_root *root, u64 dir,
   * see if there is room in the item to insert this
   * name
   */
- data_size = sizeof(*di) + name_len + sizeof(struct btrfs_item);
+ data_size = sizeof(*di) + name_len;
  leaf = path->nodes[0];
  slot = path->slots[0];
  if (data_size + btrfs_item_size_nr(leaf, slot) +
-- 
1.7.1